File: wrapper-js/scripts/verify-binaries.mjs
import { accessSync, constants, existsSync } from 'node:fs';
import { dirname, join, resolve } from 'node:path';
import { fileURLToPath } from 'node:url';
import { targets } from './targets.mjs';

const __dirname = dirname(fileURLToPath(import.meta.url));
const root = resolve(__dirname, '..');

for (const target of targets) {
  const filePath = join(root, 'bin', target.name, target.filename);
  if (!existsSync(filePath)) {
    throw new Error(`Missing bundled binary for ${target.name}: ${filePath}`);
  }
  if (!target.filename.endsWith('.exe')) {
    accessSync(filePath, constants.X_OK);
  }
}

File: wrapper-py/scripts/verify_binaries.py
from __future__ import annotations

import os
from pathlib import Path


ROOT = Path(__file__).resolve().parent.parent
TARGETS = [
    ("darwin-arm64", "agentcode"),
    ("darwin-x64", "agentcode"),
    ("linux-x64", "agentcode"),
    ("win32-x64", "agentcode.exe"),
]


def main() -> None:
    for folder, filename in TARGETS:
        path = ROOT / "src" / "agentcode" / "bin" / folder / filename
        if not path.exists():
            raise SystemExit(f"missing bundled binary for {folder}: {path}")
        if not filename.endswith(".exe") and not os.access(path, os.X_OK):
            raise SystemExit(f"bundled binary is not executable: {path}")


if __name__ == "__main__":
    main()
